### ISSUE TITLE
Protect stdout/stderr logs from being clobbered

### ISSUE THREAD

**Issue by annawoodard** [Parsl/parsl#538]
Allowing users to specify explicit log names is of limited value. Most people don't care what the log name is and don't want the extra work of specifying the name. I prefer providing the user a toggle to turn logging on or off, then we name the logs in a way they can be tied easily to the task that produced them (say `task_id_app_name.stderr`). Otherwise, we have to pick between two bad options:
1) We open logs with `'w'`, and overwrite clobbered logs.
2) We open logs with `'a'`, and append to clobbered logs-- this isn't very readable or easy for users to tie logs to tasks.

Note that users do not have access to the task id in their code, so any scheme they come up with for naming their logs necessarily needs an additional mapping to recover the task ID.

**Comment by annawoodard** [comment]
Great point @benclifford. I still think that it's a problem that you see "Task 42 failed" in the parsl log but there's no way to find the stderr associated with task 42-- that has been an issue for me in workflows I've developed. I agree with you that an acceptable workaround would be that we just append the task ID to whatever name they give us i.e. `bundle_1.txt -> bundle_1_task_42.txt`.

**Comment by gordonwatts** [comment]
The head version has a different behavior compared to the 0.6.1 that is currently "stable". In particular it now appends text to stderr and stdout logs, even if you name them.

This is problematic mostly because of the way you have written the tutorial. There you talk about naming stderr and stdout to name the outputs. And you then talk about reading back in the file in those tutorial examples. As a result of reading that tutorial I then started following that pattern when I needed to look at the output of a program I ran to scan for errors, etc.

So.. what is the recommended pattern when you want to be able to grab stdout/stderr from a bash run and scan it in your workflow after the command has finished?
